Add tests for SDK extras and fix `execute`

wfchandler opened the pull request:
We've been relying on the CLI disk import tests to cover the `extras` disk importer in the SDK. However, the CLI does not use the `execute` method, and we missed that this was failing immediately due to the cancellation sender being dropped before the request could awaited.
      
Add tests for `extras` to validate that both options work. Also, add a method to DiskImport that does not accept cancellation and use that in `execute`.

Closes #930

wfchandler committed:
Don't cancel uploads run without control

Currently `execute` will drop the import handle immediately,
inadvertently canceling the request.

Leak the handle, which will never be used, so that we allow the request
to run to completion.

Closes #930

ahl commented on sdk/src/extras/disk.rs:
```diff
@@ -107,7 +108,12 @@ pub mod builder {
+            let (fut, handle) = self.execute_with_control()?;
+
+            // Leak the handle so we don't cancel the request by dropping it.
+            mem::forget(handle);
```
this seems a bit inelegant. Did we consider other options such as a `run_without_cancel` function, ignoring cancelation from `cancel_rx` in `run_with_cancel` or returning a new `Future` here that holds onto the handle so that it isn't dropped.

wfchandler replied:
That's fair, I've added a `run` method that doesn't take a cancel channel.

ahl commented:
Also: worth adding a test here?

wfchandler committed:
Add tests for SDK extras and fix `execute`

We've been relying on the CLI disk import tests to cover the `extras`
disk importer in the SDK. However, the CLI does not use the `execute`
method, and we missed that this was failing immediately due to the
cancellation sender being dropped before the request could awaited.

Add tests for `extras` to validate that both options work. Also, add a
method to DiskImport that does not accept cancellation and use that in
`execute`.

wfchandler commented:
> Also: worth adding a test here?

Yes, great point! I've added a high level test for this.

ahl commented on sdk/tests/test_extra.rs:
```diff
@@ -0,0 +1,135 @@
+use httpmock::MockServer;
```
Add the license and copyright headers?

wfchandler replied:
Good catch, thanks

wfchandler committed:
Add copyright info

wfchandler commented:
Rebased to resolve merge conflict.
